client/rest/src/test/java/org/elasticsearch/client/SyncResponseListenerTests.java

```diff
@@ -37,6 +41,25 @@
 import static org.junit.Assert.fail;
 
 public class SyncResponseListenerTests extends RestClientTestCase {
+    static void assertExceptionStackContainsCallingMethod(Exception e) {
+        boolean foundMyMethod = false;
+        // 0 is getStackTrace
+        // 1 is this method
+        // 2 is the caller, what we want
+        StackTraceElement myMethod = Thread.currentThread().getStackTrace()[2];
+        for (StackTraceElement se : e.getStackTrace()) {
+            if (se.getClassName().equals(myMethod.getClassName())
+                    && se.getMethodName().equals(myMethod.getMethodName())) {
+                foundMyMethod = true;
+                break;
+            }
+        }
+        if (false == foundMyMethod) {
+            StringWriter stack = new StringWriter();
+            e.printStackTrace(new PrintWriter(stack));
+            fail("didn't find my stack trace (looks like " + myMethod + ") in\n" + stack);
```

Comment:
I'll reword and add javadoc to the method to make it more clear why we call it. I don't want to write a whole paragraph in the exception message because I think it'd be hard to read. But I think folks will see the method's javadoc.